chia/pools/pool_wallet.py

```diff
@@ -0,0 +1,857 @@
+
+        if tx_record is None:
+            raise ValueError("failed to generate transaction to farm to pool")
+
+        await self.standard_wallet.push_transaction(tx_record)
+        return tx_record
+
+    async def _try_to_self_pool(self, target_state: PoolState):
+        pass
+
+    async def transition(self, target_state: PoolState):
+        """
+        Attempt to move to `target_state` or to LEAVING_POOL state, as appropriate.
+        Compare target_state and self.get_current_state() to determine next state
+        """
+        if target_state is None:
+            raise ValueError(f"invalid target_state {target_state}")
+        current_state = await self.get_current_state()
+
+        #
+
+        if current_state == target_state:
+            self.target_state = None
+            self.log.info("Asked to change to current state. Target = {target_state}")
+            return
+
+        if self.target_state is not None:
+            raise ValueError(
+                f"Cannot change to state {target_state} when already having target state: {self.target_state}"
+            )
+
+        # TODO: Do not set target state until all checks are done
+        self.target_state = target_state
+
+        # all_sks = [self.wallet_state_manager.private_key]
+        # owner_sk: PrivateKey = await find_owner_sk(all_sks, current_state.current.owner_pubkey)
+
+        # Check if we can join a pool (timelock)
+        # Create the first blockchain transaction
+        # Whenever we detect a new peak, potentially initiate the second blockchain transaction
+        # verify target state
+
+        tx = None
+        if target_state.state == FARMING_TO_POOL:
+            tx = await self._try_to_farm_to_pool(target_state)
+        elif target_state.state == SELF_POOLING:
+            tx = await self._try_to_self_pool(target_state)
+
+        return tx
+
+    async def join_pool(self, target_state: PoolState):
+        if target_state.state != FARMING_TO_POOL:
+            raise ValueError(f"join_pool must be called with target_state={FARMING_TO_POOL} (FARMING_TO_POOL)")
+        if self.target_state is not None:
+            raise ValueError(f"Cannot join a pool when already having target state: {self.target_state}")
+
+        return await self.transition(target_state)
+
+    async def self_pool(self, target_state: PoolState):
+        if self.target_state is not None:
+            raise ValueError(f"Cannot self pool when already having target state: {self.target_state}")
+        self.target_state = target_state
+        # current_state = await self.get_current_state()
+        # all_sks = [self.wallet_state_manager.private_key]
+        # owner_sk: PrivateKey = await find_owner_sk(all_sks, current_state.current.owner_pubkey)
+        # Check if we can self pool (timelock)
+        # Create the first blockchain transaction
+        # Whenever we detect a new peak, potentially initiate the second blockchain transaction
+        pass
```

Comment:
Also not necessary. 
There's of course a lot of commented code here overall, and jumps between `return None` and just `return` again :^)